File: apps/api/schemas/session.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SessionCreate(BaseModel):
    name: str = "Untitled Session"


class SessionUpdate(BaseModel):
    name: Optional[str] = None
    status: Optional[str] = None
    last_algorithm: Optional[str] = None


class SessionResponse(BaseModel):
    id: str
    name: str
    user_id: str
    status: str
    last_algorithm: Optional[str]
    created_at: str
    updated_at: str

    class Config:
        from_attributes = True

File: types/session.ts
export type SessionStatus =
  | "idle"
  | "thinking"
  | "writing"
  | "running"
  | "fixing"
  | "done"
  | "error";

export interface Session {
  id: string;
  name: string;
  user_id: string;
  status: SessionStatus;
  last_algorithm: string | null;
  created_at: string;
  updated_at: string;
}

export interface SessionCreate {
  name: string;
}

export interface SessionUpdate {
  name?: string;
  status?: SessionStatus;
  last_algorithm?: string;
}
